JSIL.Libraries/Includes/Bootstrap/Core/Classes/System.Delegate.js

```diff
@@ -109,35 +109,7 @@ JSIL.ImplementExternals("System.Delegate", function ($) {
       if (typeof (delegatePublicInterface.New) !== "function")
         JSIL.Host.abort(new Error("Invalid delegate type"));
 
-      if (!method.IsStatic && firstArgument == null && delegateType.GetMethod("Invoke").GetParameters().length == method.GetParameters().length + 1) {
-        //TODO: Check that first delegate Invoke parameter is assignable to method.DeclaringType
-        return delegatePublicInterface.New(
-          null,
-          function () {
-            var realFirstAgument = arguments[0];
-            var impl = JSIL.$GetMethodImplementation(method, firstArgument);
-
-            if (!impl) {
-              throw new System.Exception("Method has no implementation");
-            } else if (typeof (impl) === "function") {
-              var f = delegatePublicInterface.New(realFirstAgument, impl, function () { return null; });
-              return f.apply(null, Array.prototype.slice.call(arguments, 1));
-            } else {
-              throw new Error("Unexpected JSIL.$GetMethodImplementation result");
-            }
-          },
-          function () { return method; });
-      }
-
-      var impl = JSIL.$GetMethodImplementation(method, firstArgument);
-
-      if (!impl) {
-        throw new System.Exception("Method has no implementation");
-      } else if (typeof (impl) === "function") {
-        return delegatePublicInterface.New(firstArgument, impl, function () { return method; });
-      } else {
-        throw new Error("Unexpected JSIL.$GetMethodImplementation result");
-      }
+      return delegatePublicInterface.New(firstArgument, null, new JSIL.MethodInfoMethodPointerInfo(method));
```

Comment:
Wow, MethodInfoMethodPointerInfo sure is a name.

A method pointer constructed from a MethodInfo, I assume?